### ISSUE TITLE
ServiceBusTriggerAttribute and BrokeredMessage

### ISSUE THREAD

**Issue by lukepuplett** [Azure/azure-webjobs-sdk#2298]
Hello,

I was writing my first queue triggered function using "Functions 2" and I wanted to get access to the `UserProperties` on the message.

While reading, I have become confused by the **ServiceBusTriggerAttribute**. I'm using one from here:

```
#region Assembly Microsoft.Azure.WebJobs.ServiceBus, Version=3.0.6.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35
// C:\Users\LukePuplett\.nuget\packages\microsoft.azure.webjobs.extensions.servicebus\3.0.6\lib\netstandard2.0\Microsoft.Azure.WebJobs.ServiceBus.dll
#endregion
```

The XML doc comments says.

> The method parameter type can be one of the following

and goes on to list, `BrokeredMessage`, `string`, `byte[]` or a model type that JSON will layout into.

![image](https://user-images.githubusercontent.com/5802524/62886944-192f7380-bd34-11e9-88dd-212eafcbb333.png)

But the NuGet package 3.0.6 seems to depend on **Microsoft.Azure.ServiceBus 4.0.0** which does not have a **BrokeredMessage** type which is familiar from docs and blogs I've been reading recently.

![image](https://user-images.githubusercontent.com/5802524/62887213-bb4f5b80-bd34-11e9-922c-e5cff5182b16.png)

But it does have a **Message** type.

![image](https://user-images.githubusercontent.com/5802524/62887231-c4d8c380-bd34-11e9-9d3b-c6cc637d33cb.png)

**Are we in some kind of transition stage and the docs are all slow to catch-up?**

---

Just as a side, as a developer with access to the source code here, I took a couple of hours to try to figure it all out myself, but finding active source repo for a particular version of a type in a package, the branch its in and then tracing it to its dependent packages and repos seems like a Gordian Knot problem.

The complexity seems unassailable. I feel stupid. Is this something you'd expect complete outsiders to be able to make sense of or would you have to work closely in the team in Redmond for a while to know how it all fits together - is there still a lot of information held only in the minds of the maintainers?

**Comment by DAllanCarr** [comment]
Check versions. 

> These parameters are for Azure Functions version 1.x; for 2.x, use Message instead of BrokeredMessage.

**Comment by lukepuplett** [comment]
Thanks. I see it, buried 15 lines up. It at least needs a purple Note callout. These binding pages have too much responsibility and need refactoring; e.g. the Cosmos DB v2 page is enormous.

Azure docs are generally excellent. However, the Function docs are hard to follow.

I guess Functions has been under rapid change and the docs have suffered during this growth, and they're having to deal with all the triggers and bindings in all the supported languages and version changes.
